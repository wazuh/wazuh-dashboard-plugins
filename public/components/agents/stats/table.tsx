/*
 * Wazuh app - Component to display the table in the Agent stats
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React from 'react';
import {
  EuiButtonEmpty,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiIcon,
  EuiInMemoryTable,
  EuiLoadingSpinner,
  EuiPanel,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import * as FileSaver from '../../../services/file-saver';
import { formatUIDate } from '../../../react-services';
import {
  UI_ERROR_SEVERITIES,
  UIErrorLog,
  UIErrorSeverity,
  UILogLevel,
} from '../../../react-services/error-orchestrator/types';
import { UI_LOGGER_LEVELS } from '../../../../common/constants';
import { getErrorOrchestrator } from '../../../react-services/common-services';

export function AgentStatTable({ columns, title, start, end, loading, items, exportCSVFilename }) {
  return (
    <EuiPanel>
      <EuiFlexGroup justifyContent="spaceBetween">
        <EuiFlexItem grow={false}>
          <EuiText>{title}</EuiText>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiText>
            <EuiIcon type="calendar" /> Start:{' '}
            {loading ? <EuiLoadingSpinner size="s" /> : start ? formatUIDate(start) : '-'} - End:{' '}
            {loading ? <EuiLoadingSpinner size="s" /> : end ? formatUIDate(end) : '-'}
          </EuiText>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiHorizontalRule margin="xs" />
      <EuiInMemoryTable columns={columns} items={items || []} loading={loading} pagination={true} />
      <EuiSpacer size="xs" />
      <EuiFlexGroup justifyContent="flexEnd">
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            onClick={() => downloadCsv(columns, items, exportCSVFilename)}
            iconType="importAction"
            isDisabled={loading}
          >
            Download CSV
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPanel>
  );
}

function downloadCsv(columns: any[], data: any[], filename: string) {
  try {
    const header = columns.map((column) => column.name).join(',');
    const body = data.map((row) => columns.map((column) => row[column.field]).join(',')).join('\n');
    const result = [header, body].join('\n');
    const blob = new Blob([result], { type: 'text/csv' }); // eslint-disable-line
    FileSaver.saveAs(blob, `${filename}.csv`);
  } catch (error) {
    const options: UIErrorLog = {
      context: 'downloadCsv',
      level: UI_LOGGER_LEVELS.ERROR as UILogLevel,
      severity: UI_ERROR_SEVERITIES.BUSINESS as UIErrorSeverity,
      error: {
        error: error,
        message: `Error generating CSV: ${error.message}`,
        title: `CSV: ${error.name}`,
      },
    };
    getErrorOrchestrator().handleError(options);
  }
}
