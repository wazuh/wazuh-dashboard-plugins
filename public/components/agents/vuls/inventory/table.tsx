/*
 * Wazuh app - Agent vulnerabilities table component
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { Component } from 'react';
import { Direction } from '@elastic/eui';
import { FlyoutDetail } from './flyout';
import { filtersToObject, IFilter, IWzSuggestItem } from '../../../wz-search-bar';
import { TableWzAPI } from '../../../../components/common/tables';
import { getFilterValues } from './lib';
import { formatUIDate } from '../../../../react-services/time-service';

export class InventoryTable extends Component {
  state: {
    error?: string;
    pageIndex: number;
    pageSize: number;
    sortField: string;
    isFlyoutVisible: Boolean;
    sortDirection: Direction;
    isLoading: boolean;
    currentItem: {};
  };

  suggestions: IWzSuggestItem[] = [
    {
      type: 'q',
      label: 'name',
      description: 'Filter by package ID',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('name', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'cve',
      description: 'Filter by CVE ID',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('cve', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'version',
      description: 'Filter by CVE version',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('version', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'architecture',
      description: 'Filter by architecture',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('architecture', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'severity',
      description: 'Filter by Severity',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('severity', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'cvss2_score',
      description: 'Filter by CVSS2',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('cvss2_score', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'cvss3_score',
      description: 'Filter by CVSS3',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('cvss3_score', value, this.props.agent.id),
    },
    {
      type: 'q',
      label: 'detection_time',
      description: 'Filter by Detection Time',
      operators: ['=', '!=', '~'],
      values: async (value) => getFilterValues('detection_time', value, this.props.agent.id),
    },
  ];

  props!: {
    filters: IFilter[];
    agent: any;
    items: [];
    onFiltersChange: Function;
  };

  constructor(props) {
    super(props);

    this.state = {
      pageIndex: 0,
      pageSize: 15,
      sortField: 'name',
      sortDirection: 'asc',
      isLoading: false,
      isFlyoutVisible: false,
      currentItem: {},
    };
  }

  closeFlyout() {
    this.setState({ isFlyoutVisible: false, currentItem: {} });
  }

  async showFlyout(item, redirect = false) {
    //if a flyout is opened, we close it and open a new one, so the components are correctly updated on start.
    this.setState({ isFlyoutVisible: false }, () =>
      this.setState({ isFlyoutVisible: true, currentItem: item })
    );
  }

  async componentDidUpdate(prevProps) {
    const { filters } = this.props;
    if (JSON.stringify(filters) !== JSON.stringify(prevProps.filters)) {
      this.setState({ pageIndex: 0, isLoading: true });
    }
  }

  buildSortFilter() {
    const { sortField, sortDirection } = this.state;
    const direction = sortDirection === 'asc' ? '+' : '-';

    return direction + sortField;
  }

  buildFilter() {
    const { pageIndex, pageSize } = this.state;
    const filters = filtersToObject(this.props.filters);
    const filter = {
      ...filters,
      offset: pageIndex * pageSize,
      limit: pageSize,
      sort: this.buildSortFilter(),
    };
    return filter;
  }

  columns() {
    let width;
    (((this.props.agent || {}).os || {}).platform || false) === 'windows'
      ? (width = '60px')
      : (width = '80px');
    return [
      {
        field: 'name',
        name: 'Name',
        sortable: true,
        width: '100px',
      },
      {
        field: 'version',
        name: 'Version',
        sortable: true,
        truncateText: true,
        width: `${width}`,
      },
      {
        field: 'architecture',
        name: 'Architecture',
        sortable: true,
        width: '100px',
      },
      {
        field: 'severity',
        name: 'Severity',
        sortable: true,
        width: `${width}`,
      },
      {
        field: 'cve',
        name: 'CVE',
        sortable: true,
        truncateText: true,
        width: `${width}`,
      },
      {
        field: 'cvss2_score',
        name: 'CVSS2 Score',
        sortable: true,
        width: `${width}`,
      },
      {
        field: 'cvss3_score',
        name: 'CVSS3 Score',
        sortable: true,
        width: `${width}`,
      },
      {
        field: 'detection_time',
        name: 'Detection Time',
        sortable: true,
        width: `100px`,
        render: formatUIDate,
      },
    ];
  }

  renderTable() {
    const getRowProps = (item) => {
      const id = `${item.name}-${item.cve}-${item.architecture}-${item.version}-${item.severity}-${item.cvss2_score}-${item.cvss3_score}-${item.detection_time}`;
      return {
        'data-test-subj': `row-${id}`,
        onClick: () => this.showFlyout(item),
      };
    };

    const { error } = this.state;
    const { filters, onFiltersChange } = this.props;
    const columns = this.columns();
    const selectFields = `select=${[
      'cve',
      'architecture',
      'version',
      'name',
      'severity',
      'cvss2_score',
      'cvss3_score',
      'detection_time',
      'title',
      'condition',
      'updated',
      'published',
      'external_references'
    ].join(',')}`;

    return (
      <TableWzAPI
        title="Vulnerabilities"
        tableColumns={columns}
        tableInitialSortingField="name"
        searchTable={true}
        searchBarSuggestions={this.suggestions}
        endpoint={`/vulnerability/${this.props.agent.id}?${selectFields}`}
        isExpandable={true}
        rowProps={getRowProps}
        mapResponseItem={(item) => ({
          ...item,
          // Some vulnerability data could not contain the external_references field.
          // This causes the rendering of them can crash when opening the flyout with the details.
          // So, we ensure the fields are defined with the expected data structure.
          external_references: Array.isArray(item?.external_references) 
            ? item?.external_references
            : []
        })}
        error={error}
        downloadCsv={true}
        filters={filters}
        onFiltersChange={onFiltersChange}
        tablePageSizeOptions={[10, 25, 50, 100]}
      />
    );
  }

  render() {
    const table = this.renderTable();
    return (
      <div className="wz-inventory">
        {table}
        {this.state.isFlyoutVisible && (
          <FlyoutDetail
            vulName={this.state.currentItem.cve}
            agentId={this.props.agent.id}
            item={this.state.currentItem}
            closeFlyout={() => this.closeFlyout()}
            type="vulnerability"
            view="inventory"
            showViewInEvents={true}
            outsideClickCloses={true}
            {...this.props}
          />
        )}
      </div>
    );
  }
}
