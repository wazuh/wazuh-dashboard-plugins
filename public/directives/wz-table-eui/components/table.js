/*
 * Wazuh app - React base component for building tables.
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EuiBasicTable } from '@elastic/eui';

export class BasicTable extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: this.props.items,
      pageIndex: this.props.pageIndex,
      pageSize: 10,
      sortField: this.props.initialSortField || this.props.columns[0].field,
      sortDirection: 'desc'
    };

    this.lastSorting = {
      sortField: this.props.initialSortField || this.props.columns[0].field,
      sortDirection: 'desc'
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({
      items: nextProps.items,
      pageIndex: nextProps.pageIndex
    });
  }

  async onTableChange({ page = {}, sort = {} }) {
    const { index: pageIndex, size: pageSize } = page;
    const { field: sortField, direction: sortDirection } = sort;

    const sortingChanged =
      this.lastSorting.sortField !== sortField ||
      this.lastSorting.sortDirection !== sortDirection;

    if (sortingChanged) {
      this.lastSorting = { sortField, sortDirection };
      await this.props.sortByField(sortField);
    }

    const { pageOfItems } = this.filterItems(pageIndex, pageSize);
    let needMoreData = pageOfItems.includes(null);

    while (needMoreData) {
      const { items } = this.state;
      const nonNull = items.filter(item => !!item).length;
      const newItems = await this.props.getData({ offset: nonNull });
      await this.promisedSetState({
        items: newItems
      });
      const result = this.filterItems(pageIndex, pageSize);
      needMoreData = result.pageOfItems.includes(null);
    }

    this.setState({
      pageIndex,
      pageSize,
      sortField,
      sortDirection
    });
  }

  promisedSetState(newState) {
    return new Promise(resolve => {
      this.setState(newState, () => {
        resolve();
      });
    });
  }

  filterItems(pageIndex, pageSize) {
    const { items } = this.state;
    const pages = [];
    const len = items.length;
    for (let i = 0, j = len; i < j; i += pageSize) {
      pages.push(items.slice(i, i + pageSize));
    }
    return { pageOfItems: pages[pageIndex] || [], totalItemCount: len };
  }

  render() {
    const { pageIndex, pageSize, sortField, sortDirection } = this.state;

    const { pageOfItems, totalItemCount } = this.filterItems(
      pageIndex,
      pageSize
    );

    const columns = [...this.props.columns];

    const pagination = {
      pageIndex,
      pageSize,
      totalItemCount,
      pageSizeOptions: [10, 20, 50],
      hidePerPageOptions: this.state.items.length <= 10
    };

    const sorting = {
      sort: {
        field: sortField,
        direction: sortDirection
      }
    };

    return (
      <div>
        <EuiBasicTable
          sorting={sorting}
          items={pageOfItems}
          columns={columns}
          pagination={pagination}
          onChange={obj => this.onTableChange(obj)}
        />
      </div>
    );
  }
}

BasicTable.propTypes = {
  items: PropTypes.array
};
