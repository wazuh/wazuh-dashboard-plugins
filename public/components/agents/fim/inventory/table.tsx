/*
 * Wazuh app - Integrity monitoring table component
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { Component } from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiBasicTable,
  Direction,
  EuiOverlayMask,
} from '@elastic/eui';
import { WzRequest } from '../../../../react-services/wz-request';
import { FlyoutDetail } from './flyout';
import './inventory.less';
import { ICustomBadges } from '../../../wz-search-bar/components';
import { filtersToObject } from '../../../wz-search-bar';

export class InventoryTable extends Component {
  state: {
    syscheck: []
    pageIndex: number
    pageSize: number
    totalItems: number
    sortField: string
    isFlyoutVisible: Boolean
    sortDirection: Direction
    isLoading: boolean
    currentFile: {
      file: string
    }
  };

  props!: {
    filters: []
    onFilterSelect(): void
    customBadges: ICustomBadges[]
    agent: any
    items: []
    totalItems: number
  }

  constructor(props) {
    super(props);

    this.state = {
      syscheck: props.items,
      pageIndex: 0,
      pageSize: 15,
      totalItems: 0,
      sortField: 'file',
      sortDirection: 'asc',
      isLoading: false,
      isFlyoutVisible: false,
      currentFile: {
        file: ""
      }
    }
  }

  async componentDidMount() {
    const regex = new RegExp('file=' + '[^&]*');
    const match = window.location.href.match(regex);
    this.setState({totalItems: this.props.totalItems});
    if (match && match[0]) {
      const file = match[0].split('=')[1];
      this.showFlyout(decodeURIComponent(file), true);
    }
  }

  closeFlyout() {
    this.setState({ isFlyoutVisible: false, currentFile: {} });
  }

  async showFlyout(file, redirect = false) {
    let fileData = false;
    if (!redirect) {
      fileData = this.state.syscheck.filter(item => {
        return item.file === file;
      })
    } else {
      const response = await WzRequest.apiReq('GET', `/syscheck/${this.props.agent.id}`, { 'file': file });
      fileData = ((response.data || {}).data || {}).items || [];
    }
    if (!redirect)
      window.location.href = window.location.href += `&file=${file}`;
    //if a flyout is opened, we close it and open a new one, so the components are correctly updated on start.
    this.setState({ isFlyoutVisible: false }, () => this.setState({ isFlyoutVisible: true, currentFile: fileData[0] }));
  }

  componentDidUpdate(prevProps) {
    const { filters, customBadges } = this.props;
    if (JSON.stringify(filters) !== JSON.stringify(prevProps.filters) 
     || JSON.stringify(customBadges) !== JSON.stringify(prevProps.customBadges)) {
      this.setState({ pageIndex: 0, isLoading: true }, this.getSyscheck)
    }
  }

  async getSyscheck() {
    const agentID = this.props.agent.id;

    const syscheck = await WzRequest.apiReq(
      'GET',
      `/syscheck/${agentID}`,
      this.buildFilter()
    );

    this.setState({
      syscheck: (((syscheck || {}).data || {}).data || {}).items || {},
      totalItems: (((syscheck || {}).data || {}).data || {}).totalItems - 1,
      isLoading: false
    });
  }

  buildSortFilter() {
    const { sortField, sortDirection } = this.state;

    const field = (sortField === 'os_name') ? '' : sortField;
    const direction = (sortDirection === 'asc') ? '+' : '-';

    return direction + field;
  }

  buildFilter() {
    const { pageIndex, pageSize } = this.state;
    const filters = filtersToObject(this.props.filters);
    const filter = {
      ...filters,
      offset: pageIndex * pageSize,
      limit: pageSize,
      sort: this.buildSortFilter(),
      type: 'file'
    };
    return filter;
  }

  onTableChange = ({ page = {}, sort = {} }) => {
    const { index: pageIndex, size: pageSize } = page;
    const { field: sortField, direction: sortDirection } = sort;
    this.setState({
      pageIndex,
      pageSize,
      sortField,
      sortDirection,
      isLoading: true,
    },
      () => this.getSyscheck()
    );
  };

  columns() {
    let width;
    this.props.agent.os.platform === 'windows' ? width = '60px' : width = '80px';
    return [
      {
        field: 'file',
        name: 'File',
        sortable: true,
        width: '250px'
      },
      {
        field: 'mtime',
        name: 'Last Modified',
        sortable: true,
        width: '100px'
      },
      {
        field: 'uname',
        name: 'User',
        sortable: true,
        truncateText: true,
        width: `${width}`
      },
      {
        field: 'uid',
        name: 'User ID',
        sortable: true,
        truncateText: true,
        width: `${width}`
      },
      {
        field: 'gname',
        name: 'Group',
        sortable: true,
        truncateText: true,
        width: `${width}`
      },
      {
        field: 'gid',
        name: 'Group ID',
        sortable: true,
        truncateText: true,
        width: `${width}`
      },
      {
        field: 'perm',
        name: 'Permissions',
        sortable: true,
        truncateText: true,
        width: `${width}`
      },
      {
        field: 'size',
        name: 'Size',
        sortable: true,
        width: `${width}`
      }
    ]
  }

  renderFilesTable() {
    const getRowProps = item => {
      const { file } = item;
      return {
        'data-test-subj': `row-${file}`,
        onClick: () => this.showFlyout(file),
      };
    };

    const { syscheck, pageIndex, pageSize, totalItems, sortField, sortDirection, isLoading } = this.state;
    const columns = this.columns();
    const pagination = {
      pageIndex: pageIndex,
      pageSize: pageSize,
      totalItemCount: totalItems,
      pageSizeOptions: [15, 25, 50, 100],
    }
    const sorting = {
      sort: {
        field: sortField,
        direction: sortDirection,
      },
    };

    return (
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiBasicTable
            items={syscheck}
            columns={columns}
            pagination={pagination}
            onChange={this.onTableChange}
            rowProps={getRowProps}
            sorting={sorting}
            itemId="file"
            isExpandable={true}
            loading={isLoading}
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }

  render() {
    const filesTable = this.renderFilesTable();
    return (
      <div>
        {filesTable}
        {this.state.isFlyoutVisible &&
          <EuiOverlayMask
            onClick={(e: Event) => { e.target.className === 'euiOverlayMask' && this.closeFlyout() }} >
            <FlyoutDetail
              fileName={this.state.currentFile.file}
              agentId={this.props.agent.id}
              closeFlyout={() => this.closeFlyout()}
              type='file'
              view='inventory'
              showViewInEvents={true}
              {...this.props} />
          </EuiOverlayMask>
        }
      </div>
    )
  }
}
