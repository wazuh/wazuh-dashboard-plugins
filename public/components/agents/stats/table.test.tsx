/*
 * Wazuh app - React test for Ruleset component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import { mount } from 'enzyme';
import { AgentStatTable } from './table';

jest.mock('../../../kibana-services', () => ({
  getAngularModule: jest.fn(),
  getHttp: () => ({
    basePath: {
      prepend: (str) => str,
    },
  }),
}));
jest.mock(
  '../../../../../../node_modules/@elastic/eui/lib/services/accessibility/html_id_generator',
  () => ({
    htmlIdGenerator: () => () => 'htmlId',
  })
);

const tableColumns = [
  {
    field: 'location',
    name: 'Location',
    sortable: true,
  },
  {
    field: 'events',
    name: 'Events',
    sortable: true,
  },
  {
    field: 'bytes',
    name: 'Bytes',
    sortable: true,
  },
];

describe('AgentStatTable component', () => {
  it('Renders correctly to match the snapshot', () => {
    const wrapper = mount(
      <AgentStatTable
        columns={tableColumns}
        loading={false}
        start={''}
        end={''}
        title="Test"
        items={[]}
        exportCSVFilename={`agent-stats-10101-logcollector-global`}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should return loading in process', () => {
    const wrapper = mount(
      <AgentStatTable
        columns={tableColumns}
        loading={true}
        start={''}
        end={''}
        title="Test"
        items={[]}
        exportCSVFilename={`agent-stats-10101-logcollector-global`}
      />
    );

    expect(wrapper.find('EuiLoadingSpinner').exists()).toBeTruthy();
    expect(wrapper.find('EuiLoadingSpinner').first().prop('size')).toBe('s');
  });

  it('Checking onClick event', () => {
    const wrapper = mount(
      <AgentStatTable
        columns={tableColumns}
        loading={true}
        start={''}
        end={''}
        title="Test"
        items={[]}
        exportCSVFilename={`agent-stats-10101-logcollector-global`}
      />
    );

    const mockOnClick = jest.fn();
    wrapper.find('EuiButtonEmpty').props().onClick = mockOnClick;
    // @ts-ignore
    wrapper.find('EuiButtonEmpty').props().onClick();
    expect(mockOnClick).toHaveBeenCalled();
  });
});
