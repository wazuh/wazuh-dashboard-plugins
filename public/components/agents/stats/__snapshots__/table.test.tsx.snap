// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentStatTable component Renders correctly to match the snapshot 1`] = `
<AgentStatTable
  columns={
    Array [
      Object {
        "field": "location",
        "name": "Location",
        "sortable": true,
      },
      Object {
        "field": "events",
        "name": "Events",
        "sortable": true,
      },
      Object {
        "field": "bytes",
        "name": "Bytes",
        "sortable": true,
      },
    ]
  }
  end=""
  exportCSVFilename="agent-stats-10101-logcollector-global"
  items={Array []}
  loading={false}
  start=""
  title="Test"
>
  <EuiPanel>
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <EuiFlexGroup
        justifyContent="spaceBetween"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <EuiFlexItem
            grow={false}
          >
            <div
              className="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <EuiText>
                <div
                  className="euiText euiText--medium"
                >
                  Test
                </div>
              </EuiText>
            </div>
          </EuiFlexItem>
          <EuiFlexItem
            grow={false}
          >
            <div
              className="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <EuiText>
                <div
                  className="euiText euiText--medium"
                >
                  <EuiIcon
                    type="calendar"
                  >
                    <span
                      data-euiicon-type="calendar"
                    />
                  </EuiIcon>
                   Start:
                   
                  -
                   - End:
                   
                  -
                </div>
              </EuiText>
            </div>
          </EuiFlexItem>
        </div>
      </EuiFlexGroup>
      <EuiHorizontalRule
        margin="xs"
      >
        <hr
          className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
        />
      </EuiHorizontalRule>
      <EuiInMemoryTable
        columns={
          Array [
            Object {
              "field": "location",
              "name": "Location",
              "sortable": true,
            },
            Object {
              "field": "events",
              "name": "Events",
              "sortable": true,
            },
            Object {
              "field": "bytes",
              "name": "Bytes",
              "sortable": true,
            },
          ]
        }
        items={Array []}
        loading={false}
        pagination={true}
        responsive={true}
        tableLayout="fixed"
      >
        <EuiBasicTable
          columns={
            Array [
              Object {
                "field": "location",
                "name": "Location",
                "sortable": true,
              },
              Object {
                "field": "events",
                "name": "Events",
                "sortable": true,
              },
              Object {
                "field": "bytes",
                "name": "Bytes",
                "sortable": true,
              },
            ]
          }
          items={Array []}
          loading={false}
          noItemsMessage={
            <EuiI18n
              default="No items found"
              token="euiBasicTable.noItemsMessage"
            />
          }
          onChange={[Function]}
          pagination={
            Object {
              "hidePerPageOptions": undefined,
              "pageIndex": 0,
              "pageSize": 10,
              "pageSizeOptions": Array [
                10,
                25,
                50,
              ],
              "totalItemCount": 0,
            }
          }
          responsive={true}
          tableLayout="fixed"
        >
          <div
            className="euiBasicTable"
          >
            <div>
              <EuiTableHeaderMobile>
                <div
                  className="euiTableHeaderMobile"
                >
                  <EuiFlexGroup
                    alignItems="baseline"
                    justifyContent="spaceBetween"
                    responsive={false}
                  >
                    <div
                      className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                    >
                      <EuiFlexItem
                        grow={false}
                      >
                        <div
                          className="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </EuiFlexItem>
                      <EuiFlexItem
                        grow={false}
                      >
                        <div
                          className="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </EuiFlexItem>
                    </div>
                  </EuiFlexGroup>
                </div>
              </EuiTableHeaderMobile>
              <EuiTable
                id="htmlId"
                responsive={true}
                tableLayout="fixed"
              >
                <table
                  className="euiTable euiTable--responsive"
                  id="htmlId"
                  tabIndex={-1}
                >
                  <EuiScreenReaderOnly>
                    <caption
                      className="euiScreenReaderOnly euiTableCaption"
                    >
                      <EuiDelayRender
                        delay={500}
                      />
                    </caption>
                  </EuiScreenReaderOnly>
                  <EuiTableHeader>
                    <thead>
                      <tr>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_location_0"
                          key="_data_h_location_0"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_location_0"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Location",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Location"
                                    >
                                      Location
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_events_1"
                          key="_data_h_events_1"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_events_1"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Events",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Events"
                                    >
                                      Events
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_bytes_2"
                          key="_data_h_bytes_2"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_bytes_2"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Bytes",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Bytes"
                                    >
                                      Bytes
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                      </tr>
                    </thead>
                  </EuiTableHeader>
                  <EuiTableBody>
                    <tbody>
                      <EuiTableRow>
                        <tr
                          className="euiTableRow"
                        >
                          <EuiTableRowCell
                            align="center"
                            colSpan={3}
                            isMobileFullWidth={true}
                          >
                            <td
                              className="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                              colSpan={3}
                              style={
                                Object {
                                  "width": undefined,
                                }
                              }
                            >
                              <div
                                className="euiTableCellContent euiTableCellContent--alignCenter"
                              >
                                <span
                                  className="euiTableCellContent__text"
                                >
                                  <EuiI18n
                                    default="No items found"
                                    token="euiBasicTable.noItemsMessage"
                                  >
                                    No items found
                                  </EuiI18n>
                                </span>
                              </div>
                            </td>
                          </EuiTableRowCell>
                        </tr>
                      </EuiTableRow>
                    </tbody>
                  </EuiTableBody>
                </table>
              </EuiTable>
            </div>
          </div>
        </EuiBasicTable>
      </EuiInMemoryTable>
      <EuiSpacer
        size="xs"
      >
        <div
          className="euiSpacer euiSpacer--xs"
        />
      </EuiSpacer>
      <EuiFlexGroup
        justifyContent="flexEnd"
      >
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <EuiFlexItem
            grow={false}
          >
            <div
              className="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <EuiButtonEmpty
                iconType="importAction"
                isDisabled={false}
                onClick={[Function]}
              >
                <button
                  className="euiButtonEmpty euiButtonEmpty--primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <EuiButtonContent
                    className="euiButtonEmpty__content"
                    iconSide="left"
                    iconSize="m"
                    iconType="importAction"
                    textProps={
                      Object {
                        "className": "euiButtonEmpty__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButtonEmpty__content"
                    >
                      <EuiIcon
                        className="euiButtonContent__icon"
                        color="inherit"
                        size="m"
                        type="importAction"
                      >
                        <span
                          className="euiButtonContent__icon"
                          color="inherit"
                          data-euiicon-type="importAction"
                          size="m"
                        />
                      </EuiIcon>
                      <span
                        className="euiButtonEmpty__text"
                      >
                        Download CSV
                      </span>
                    </span>
                  </EuiButtonContent>
                </button>
              </EuiButtonEmpty>
            </div>
          </EuiFlexItem>
        </div>
      </EuiFlexGroup>
    </div>
  </EuiPanel>
</AgentStatTable>
`;
